src/task2/Main.java

```diff
@@ -0,0 +1,62 @@
+package task2;
+
+import task2.obstacles.Obstacle;
+import task2.obstacles.Treadmill;
+import task2.obstacles.Wall;
+import task2.participant.Cat;
+import task2.participant.Human;
+import task2.participant.Participant;
+import task2.participant.Robot;
+
+public class Main {
+
+    public static void main(String[] args) {
+
+        Participant[] participants = {
+                new Human("Alex", 180, 2),
+                new Cat("Tima", 300, 4),
+                new Robot("R2D2", 500, 1)
+        };
+
+        Obstacle[] obstacles = {
+                new Treadmill("MegaTrack", 200),
+                new Wall("WallTOP", 3)
+        };
+
+        for (Participant participant : participants) {
+            System.out.print("\n");
+            boolean passedAllObstacles = true;
+            for (Obstacle obstacle : obstacles) {
+                if (obstacle instanceof Treadmill) {
+                    participant.run();
+                    if (!((Treadmill) obstacle).overcome(participant.getMaxRunDistance())) {
```

Comment:
violates the rule of polymorphism, bacaue

> Polymorphism is the ability of a program to identically use objects with the same interface without information about the specific type of this object.

 In order to fix this, it is necessary to move the implementations to the appropriate classes